### DISCUSSION TITLE
Responsibility of  different catalog.

### DISCUSSION THREAD

**Discussion by J0HN50N133** [GreptimeTeam/greptimedb#4271] [Q&A] [answered]
Currently there are two implemention of `CatalogManager`: `KvBackendCatalogManager` and `MemoryCatalogManager`. I want to know if the following understanding about their responsibility is correct： `MemoryCatalogManager` should only serve as a placeholder for `CatalogManager` in testing env or  `QueryEngine` of `RegionServer` The `KvBackendCatalogManager` should be held by the frontend to handle logic plan. Then sql text can be transformed into a substrait plan and distributed to the region server. [At this point](https://github.com/GreptimeTeam/greptimedb/blob/8e306f3f514c3998cdc322edfe85a4f011b591c0/src/datanode/src/region_server.rs#L166), the region server can simply provide the catalog_list, which is built on the spot, to datafusion for execution.

**Comment by killme2008** [comment]
Yes, I think you are correct. @waynexia Could you confirm it?

**Reply by waynexia** [reply, marked as answer]
Precisely 👍 Memory catalog is only for test and region server, and kv backend is for all other use cases.
